### PR TITLE
Convert integer data types to strings

### PR DESCRIPTION
**Pull request by rakanalh**
Addressing Paul's feedback on #3969

### CODE REVIEW

File: raiden/storage/serialization/types.py [hunk 1]
```diff
@@ -184,17 +184,17 @@ def message_event_schema_deserialization(
...
-        PaymentAmount: fields.Int,
-        PaymentID: fields.Int,
-        PaymentWithFeeAmount: fields.Int,
-        TransferID: fields.Int,
+        BlockExpiration: IntegerToStringField,
+        BlockNumber: IntegerToStringField,
+        FeeAmount: IntegerToStringField,
+        LockedAmount: IntegerToStringField,
+        BlockGasLimit: IntegerToStringField,
+        MessageID: IntegerToStringField,
```
Comment (palango) [inline raiden/storage/serialization/types.py]: here a `Int` field should be enough. Not sure if this would increase usability in migrations.

Comment (rakanalh) [reply raiden/storage/serialization/types.py]: Should we keep this for consistency (all integers are converted to strings during serialization)?

Comment (palango) [reply raiden/storage/serialization/types.py]: Fine for me

**Comment by rakanalh** [comment]
I am seeing failures such as:
```
E   ValueError: invalid literal for int() with base 10: 'None'
```
This is mostly caused by this:
https://github.com/raiden-network/raiden/blob/3fa35a4308d5a588f27ad390d10301a1875c8960/raiden/transfer/state.py#L264

Where a None value during serialization would translate to `'None'` as a string. Deserializing this and converting to `int` fails.

I have 2 options:
1. Either have the `IntegerToStringField` detect None values and return an empty string.
2. Introduce `OptionalIntegerToStringField` which handles `Optional` values.

I opted for the second one.

File: raiden/storage/serialization/fields.py [hunk 2]
```diff
@@ -18,6 +18,18 @@ def _deserialize(self, value: str, attr: Any, data: Any) -> int:
         return int(value)
 
 
+class OptionalIntegerToStringField(marshmallow.fields.Field):
```
Comment (palango) [inline raiden/storage/serialization/fields.py]: Would be nice to have some round-trip tests.

**Pull request merged**